examples/examples/fmt-custom-field.rs

```diff
@@ -0,0 +1,36 @@
+//! This example demonstrates overriding the way `tracing-subscriber`'s
+//! `FmtSubscriber` formats fields on spans and events, using a closure.
+#![deny(rust_2018_idioms)]
+use tracing_subscriber;
+
+#[path = "fmt/yak_shave.rs"]
+mod yak_shave;
+
+fn main() {
+    use tracing_subscriber::{fmt::format, prelude::*};
+
+    // Format fields using the provided closure.
+    let format = format::debug_fn(|writer, _field, value| {
+        // We'll format _just_ the value of each field, ignoring the field name.
+        write!(writer, "{:?}", value)
+    })
+    // Separate each field with a comma.
+    // This method is provided by an extension trait in the
+    // `tracing-subscriber` prelude.
+    .delimited(", ");
```

Comment:
Nit: can you, in a comment above the `main` function, show that the before and after this change looks like? I think that'll make it a bit more concrete for folks.